src/Data/List/Relation/Unary/All.agda

```diff
@@ -135,6 +130,21 @@ module _ p {A : Set a} {P : Pred A (a ⊔ p)}
   forM : ∀ {Q : Pred A q} {xs} → All Q xs → (Q ⊆ M ∘′ P) → M (All P xs)
   forM = forA p App
 
+------------------------------------------------------------------------
+-- Generalised lookup based on a proof of Any
+
+module _ {P : Pred A p} {Q : Pred A q} {R : Pred A r}
+         (p×q⇒r : ∀[ P ⇒ Q ⇒ R ]) where
```

Comment:
Please could we have `p×q⇒r : ∀[ P ⇒ Q ⇒ R ]` in the main type as it's quite an important part of it?